src/settings.cpp

```diff
@@ -1955,8 +1971,19 @@ CommandCost CmdChangeCompanySetting(TileIndex tile, DoCommandFlag flags, uint32
  */
 uint GetSettingIndex(const SettingDesc *sd)
 {
-	assert((sd->flags & SGF_PER_COMPANY) == 0);
-	return sd - _settings;
+	assert(sd != nullptr && (sd->flags & SGF_PER_COMPANY) == 0);
+	return sd - _settings.begin();
+}
+
+/**
+ * Get the index of the company setting with this description.
+ * @param sd the setting to get the index for.
+ * @return the index of the setting to be used for CMD_CHANGE_COMPANY_SETTING.
+ */
+static uint GetCompanySettingIndex(const SettingDesc *sd)
```

Comment:
can't  we use `const SettingDesc &sd` instead?

Reply:
Because `GetSettingFromName` returns a pointer as it could not have a match and it then returns `nullptr`. Similarly `sd` is passed almost everywhere are a pointer. That's something that could be improved upon, but I'm trying very hard to not tackle everything in one PR.

Reply:
> but I'm trying very hard to not tackle everything in one PR.

And that is appreciated :D